Lesson_16/l16_h01_class.py

```diff
@@ -0,0 +1,106 @@
+"""
+HW:16[Volkov Anton].
+
+# write class for bank card.
+# Class should reflect card lifecycle, card operations etc.
+# use CVV, PIN, Name, Surname , end date, card_num as initial params
+# class should have in addition to common logic some class attributes,
+# as minimum one classmethod and
+# as minimum  one staticmethod, two or more getters/setters
+# do not forget about block if __name__ == '__main__':
+"""
+
+
+class MonoBankCard:
+    """Class for creating cards."""
+
+    PIN_MAX_SIZE = 4
+    BANK_ID = 5375
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        """Initialize class."""
+        self._card_num = card_num
+        self.__cvv = cvv
+        self.__pin = pin
+        self._name = name
+        self._surname = surname
+        self._end_date = end_date
+        self._lvl_card = 'Universal'
+
+    @property
+    def card_num(self):
+        """Get the card number."""
+        return self._card_num
+
+    @card_num.setter
+    def card_num(self, new_card_num):
+        """Set the card number."""
+        self._card_num = new_card_num
+
+    @property
+    def cvv(self):
+        """Get the CVV."""
+        return self.__cvv
+
+    @property
+    def pin(self):
+        """Get the PIN."""
+        return self.__pin
+
+    @pin.setter
+    def pin(self, new_pin):
+        """Set the PIN."""
+        if len(new_pin) == MonoBankCard.PIN_MAX_SIZE:
+            self.__pin = new_pin
+        else:
+            print(f'PIN should be {MonoBankCard.PIN_MAX_SIZE} size.')
+
+    def block_card(self):
```

Comment:
Much closed this time. Pay attention that I was expected to see not fake but some real methods.
For example for pay we could add some protected attribute balance and decrease this balance ( if possible  balance >= amount) after each successful pay. Card block state should not allow to do pay operations etc. 

Just try to think how this works in real life and with certain level of abstraction try to reflect this logic in your code.
This is great practice to improve skills of architecture planning  and abstract thinking